Additional pop up menu options

saltrafael opened the pull request:
## PR Checklist

<!-- For the checkbox formatting to work properly, make sure there are no spaces on either side of the "x" -->

Please check all that apply to this PR using "x":

- [X] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [ ] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [X] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## Fixes

Issue Number: Closes #5782

## What is the current behavior?

![](https://i.postimg.cc/zGgXf6Vr/image.png)

## What is the new behavior?

![](https://i.postimg.cc/gcQC0JRx/image.png)

On own content:
![](https://i.postimg.cc/RCHPXT60/image.png) ![](https://i.postimg.cc/Hk6Ddttg/image.png)

## Other information

just missing playlist stuff for now

<!-- If this PR contains a breaking change, please describe the impact and solution strategy for existing applications below. -->

tzarebczan commented on ui/component/claimMenuList/index.js:
```diff
@@ -1,20 +1,41 @@
 import { connect } from 'react-redux';
-import { makeSelectClaimForUri, makeSelectClaimIsMine } from 'lbry-redux';
+import { makeSelectClaimForUri, selectMyActiveClaims, makeSelectFileInfoForUri, doPrepareEdit } from 'lbry-redux';
```
I'm not sure we can use selectMyActiveClaims for this feature - this would require knowing all your claims, which we don't run anywhere on the web at least since it's an expensive operation, and could be lots of claims to store in redux. 

We should go back to using the "claim is mine" - this comes from the resolve/claim search result: ```      "is_my_output": true,
```

True means it's their claim.   This should work with both channels and claims.

saltrafael replied:
yeah that makes sense, don't know why I thought it would be better

saltrafael commented:
## Notable new changes:

-Reverted myActiveClaims

-Added some toasts for menu actions:

![](https://i.postimg.cc/Y9s0yrxq/image.png)

-Muting and Blocking toast will link to the /block_and_mute settings page, which is a way for the action to be undone since the content being muted or blocked disappears from the page:

![](https://i.postimg.cc/DfSPfmH6/image.png)

-Switched up what Mute and Block messages display on toast, to clear confusion:

![](https://i.postimg.cc/63ynQ1jq/image.png) ![](https://i.postimg.cc/GpST6xLC/image.png)

-Also changed up redux code a little to separate better Mute from Unmute, and Subscribe from Unsubscribe, like how it already works for Block and Unblock

saltrafael commented:
### Explaining my final solution to claimIsMine:

I added a new selector called makeSelectSigningIsMine, which is similar to makeSelectClaimIsMine but instead it returns true when signing_channel is mine, which then solves the issue of not returning true on the search page:

![](https://i.postimg.cc/7LR5ZS0S/image.png)

And also prevents Reposts from returning true, as your claims, which would show on the menu pop up options like "edit" or "delete", and would break trying to edit or delete the reposted content

tzarebczan commented:
By search page do you mean the lighthouse results? We can fix that by adding the --include_is_my_output parameter to claim searches and resolves. Looks like we don't pass this in a few places that we should (i.e. homepages, top search result). 

I think that's the better way of doing it so it's consistent with the previous method we use for checking ownership. 

For claims that are your own, should also hide the mute/block button if it's not muted (allow unmute/unblock if it is).

saltrafael commented:
> I think that's the better way of doing it so it's consistent with the previous method we use for checking ownership.

Hmm I think so too, that one might be it then

> For claims that are your own, should also hide the mute/block button if it's not muted (allow unmute/unblock if it is).

Hide mute/block and show unmute/unblock? why not hide both? can you even mute/block your own claims?

tzarebczan commented:
Shouldn't allow to block/mute your own, but if someone did, should allow unmuting/blocking. 

Pushed on rebased version, before playlists. I can try to take a stab at rebasing on playlists later - unless you want to:? 

Some bugs: 
- channel analytics option doesn't load the channel on the analytics page (I see it in the URL, but loads my default).
- should probably hide options that are already visible on certain pages: 
![image](https://user-images.githubusercontent.com/8120721/121239040-be4f0080-c866-11eb-9e72-5b21e227fd85.png)


Nice to have: 
- support reposts for deletion button (i.e. I have a repost on my channel page > I can menu > delete.

saltrafael commented:
> Shouldn't allow to block/mute your own, but if someone did, should allow unmuting/blocking.

✔️
![](https://i.postimg.cc/bNp4kQXp/image.png)

> Pushed on rebased version, before playlists. I can try to take a stab at rebasing on playlists later - unless you want to:?

✔️Merged playlists changes

> Some bugs:
> 
> * channel analytics option doesn't load the channel on the analytics page (I see it in the URL, but loads my default).

✔️Fixed

> * should probably hide options that are already visible on certain pages:

✔️Now only shows options that are not already on screen
![](https://i.postimg.cc/1zhGVrTK/image.png)

> 
> Nice to have:
> 
> * support reposts for deletion button (i.e. I have a repost on my channel page > I can menu > delete.

⌛ Added support for it but when trying to delete a repost (abandon claim) the claim only returns transaction_id for the original post, so the revoke fails. maybe could return --repost_transaction_id on claim resolve?

jessopb commented:
Looking better -
I think it would be better to use the query param that comes from the analytics button and use that to set the active channel on the dash component.
If you just want to ship what you have, strip the analytics item out. 
I don't think passing setActiveChanged into the SelectChannel component is the best way.
